backend/app/assets/javascripts/spree/backend/routes.js

```diff
@@ -1,27 +1,66 @@
-Spree.routes.checkouts_api = Spree.pathFor('api/checkouts')
-Spree.routes.classifications_api = Spree.pathFor('api/classifications')
-Spree.routes.option_value_search = Spree.pathFor('api/option_values')
-Spree.routes.option_type_search = Spree.pathFor('api/option_types')
-Spree.routes.orders_api = Spree.pathFor('api/orders')
-Spree.routes.product_search = Spree.pathFor('api/products')
-Spree.routes.admin_product_search = Spree.pathFor('admin/search/products')
-Spree.routes.shipments_api = Spree.pathFor('api/shipments')
-Spree.routes.checkouts_api = Spree.pathFor('api/checkouts')
-Spree.routes.stock_locations_api = Spree.pathFor('api/stock_locations')
-Spree.routes.taxon_products_api = Spree.pathFor('api/taxons/products')
-Spree.routes.taxons_search = Spree.pathFor('api/taxons')
-Spree.routes.user_search = Spree.pathFor('admin/search/users')
-Spree.routes.variants_api = Spree.pathFor('api/variants')
-Spree.routes.users_api = Spree.pathFor('api/users')
+/**
+ * @deprecated Spree.routes will be removed in a future release. Please use Spree.pathFor instead.
+ * See: https://github.com/solidusio/solidus/issues/3405
+ */
+if(typeof Proxy == "function") {
+  Spree.routes = new Proxy({
+    states_search: Spree.pathFor('api/states'),
```

Comment:
Can we extract all of this into a variable to remove duplication? In the `else` condition, you should be able to mass-assign these properties via `Object.assign`: https://developer.mozilla.org/it/docs/Web/JavaScript/Reference/Global_Objects/Object/assign.

Reply:
Done! If Proxy is available, we're rebuilding the whole Spree.routes object (to avoid triggering the deprecation). If not, it uses Object.assign to add the admin routes to the already assigned frontend routes. Take a look and let me know if that works for you!